### PR TITLE
Add tests for Reina::Controller, Reina::GitHubController and some bugfixing

### PR DESCRIPTION
**Pull request by RoxasShadow**
ref. https://github.com/honeypotio/honeypot/issues/2483

### CODE REVIEW

File: specs/controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
+require_relative 'spec_helper'
+
+describe Reina::Controller do
+  let(:params) { [1234, 'a#b', 'c#d'] }
```
Comment (Ryman) [inline specs/controller_spec.rb]: in the last pr you were taking the user input and quoting it so would the params not be like `'"a#b"'`? or did I miss a step?

File: lib/reina/github_controller.rb [hunk 2]
```diff
@@ -88,9 +88,9 @@ def repo_full_name
       payload.dig('repository', 'full_name')
     end
 
-    def reply_message(cmd)
+    def reply_message
       url = ['https://', CONFIG[:app_name_prefix], repo_name, '-', issue_number, '.herokuapp.com'].join
-      "`#{cmd}` executed.\n\nWill be deployed at #{url}"
+      "Deployed started at #{url}..."
```
Comment (Ryman) [inline lib/reina/github_controller.rb]: Deployment

File: specs/github_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        end
+
+      end
+
+      context 'an unexpected action has been taken' do
+        let(:action) { 'updated' }
+
+        before { allow(instance).to receive(:authenticate!) }
+
+        it 'close the request silently' do
```
Comment (Ryman) [inline specs/github_controller_spec.rb]: to be clear this isn't closing the issue / pr right, just ending the request?

File: lib/reina/github_controller.rb (hunk 2 above)
Comment (RoxasShadow) [reply lib/reina/github_controller.rb]: lol, thanks

File: specs/github_controller_spec.rb (hunk 3 above)
Comment (RoxasShadow) [reply specs/github_controller_spec.rb]: yes, the http request - that is, send 202 with an empty body (cfr. `Reina::Server`)

File: specs/controller_spec.rb (hunk 1 above)
Comment (RoxasShadow) [reply specs/controller_spec.rb]: Yes, but that's true only for the CLI, from which `ARGV` returns no quotes

Comment (Ryman) [reply specs/controller_spec.rb]: I mean [here](https://github.com/honeypotio/reina/blob/master/lib/reina/github_controller.rb#L48) which seems to be passing it directly to the controller unrelated to the CLI?

**Pull request merged**